src/components/Checklist/Checklist.stories.tsx

```diff
@@ -49,18 +52,22 @@ const COLUMN_CHECKBOXES = Array.from({ length: 8 })
         tooltip: { content: `Random Tooltip ${index}` },
     }));
 
-export const Checklist: StoryFn<ChecklistProps> = (args: ChecklistProps) => {
+export const Checklist = (args: ChecklistProps) => {
```

Comment:
We should keep the `StoryFn`, it sems the CheklistProps were developed in a wrong way.
The missing props could be added this way:
`export const Checklist: StoryFn<ChecklistProps & ChecklistBase> = (args: ChecklistProps & ChecklistBase)`

And add the `export` in CheckList.tsx for the `ChecklistBase` type.

Reply:
I tried following your suggestion but was still getting lint errors related to the ChecklistVertical | ChecklistHorizontal type, so I re-structured the ChecklistProps instead. Have a look.